### DISCUSSION TITLE
Add C++20 modules support

### DISCUSSION THREAD

**Discussion by kelteseth** [aws/aws-sdk-cpp#3938] [General]
### Describe the feature

Hi,
I'm not a user of aws-sdk-cpp, but I'm tracking the progress of c++20 modules support on https://arewemodulesyet.org/ and this is the issue we are linking to in our table of c++20 modules support.

### Use Case

Modules are part of C++20 and should greatly increase compile times, see https://en.cppreference.com/w/cpp/language/modules

### Acknowledgements

- [ ] I may be able to implement this feature request
- [X] This feature might incur a breaking change

**Comment by jmklix** [comment]
Thanks for opening this feature request and making the effort to get modules widely supported. We are currently blocked by curl not supporting modules yet. This is something that we would like to add support for, but we currently don't have any timeline for when this might be supported by this sdk.

**Comment by kai-ion** [comment]
Converting this to a Discussion. It's an improvements we plan to pursue over time rather than one actionable change with a near-term milestone, so a Discussion better fits collecting the specifics and design input.
